modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -8463,11 +8468,16 @@ public Set<String> getOpenAPIGeneratorIgnoreList() {
         return openapiGeneratorIgnoreList;
     }
 
+    @Override
```

Comment:
can you describe what `isTypeErasedGenerics` does with a line or two and I'll add a docstring to the method below in my upcoming PR?

Reply:
Sorry for the late reply. `isTypeErasedGenerics`  is meant to tell the code generator whether the target language is erasing the generic parameter(s) at runtime, which has implications for handling the possible types in things like the constructor or `instanceof` checks. As an example, if the language has type erased generics, the compiler throws an error when you do something like 
```java
public Constructor(final List<String> args) {...}

public Constructor(final List<Integer> args) {...}
```
because both the signatures erase to the same signature `Constructor(final List args)` which is not allowed. Therefore you need a way to check whether the code generation can include the full parametrized type `List<String>` and do so potentially multiple times, or if it has to only use it once as the base type `List`.

Reply:
what about using ` {{^vendorExtensions.x-duplicated-data-type}} ...  {{/vendorExtensions.x-duplicated-data-type}}`  instead?

https://github.com/B4ckslash/openapi-generator/blob/3702fa2a7db0fbfac3007c9e2f6eb095e0992c4e/modules/openapi-generator/src/main/resources/Java/libraries/jersey2/oneof_model.mustache#L194C1-L195C1

Reply:
Yes I use that as well, but I have to determine on the codegen side whether that is even applicable. In languages that do not erase their generics, `List<String>` and `List<Integer>` are different data types, so in those languages using `x-duplicated-data-type` makes no sense. Depending on whether `isTypeErasedGenerics` is true, I set the `x-duplicated-data-type` flag for types affected by type erasure (if there are multiple variants of it).